test/functional/feature_dirsymlinks.py

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python3
+# Copyright (c) 2022 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test successful startup with symlinked directories.
+"""
+
+import os
+import sys
+
+from test_framework.test_framework import BitcoinTestFramework, SkipTest
+
+
+def rename_and_link(*, from_name, to_name):
+    os.rename(from_name, to_name)
+    os.symlink(to_name, from_name)
+    assert os.path.islink(from_name) and os.path.isdir(from_name)
```

Comment:
This will only work with absolute paths.

```sh
$ cd /tmp
$ mkdir -p datadir/regtest/blocks

$ mv datadir/regtest/blocks datadir/regtest/newblocks
$ ln -s datadir/regtest/newblocks datadir/regtest/blocks
$ ls -l datadir/regtest/blocks
lrwxr-xr-x   datadir/regtest/blocks -> datadir/regtest/newblocks
# now /tmp/datadir/regtest/blocks is a symlink to /tmp/datadir/regtest/datadir/regtest/newblocks
```

I think it is ok, coz `self.nodes[0].datadir` is probably an absolute path now, given that the test passes.

Reply:
I verified `self.nodes[0].datadir` is absolute even if a relative one is given on the command line, like: `./test/functional/feature_dirsymlinks.py --tmpdir=foo`

Reply:
Right. It's a function for the scope of this test, it doesn't aim nor need to work for the general case.